### PR TITLE
Add `interaction` prop to Messages

### PR DESCRIPTION
**Pull request by Snazzah**
Towards the end of the [Message object structure](https://discord.com/developers/docs/resources/channel#message-object-message-structure) there is an `interaction` prop for messages that are created for interactions. There is also an undocumented `member` property inside of a [MessageInteraction](https://discord.com/developers/docs/interactions/slash-commands#messageinteraction) that only shows up when the message is created in gateway.

### CODE REVIEW

File: lib/structures/Message.js [hunk 1]
```diff
@@ -89,6 +94,24 @@ class Message extends Base {
         } else {
             this.referencedMessage = data.referenced_message;
         }
+        if(data.interaction) {
+            this.interaction = data.interaction;
```
Comment (bsian03) [inline lib/structures/Message.js]: Probably better if user/member fetching was done before assigning to `this`?

File: lib/structures/Message.js [hunk 2]
```diff
@@ -31,6 +31,11 @@ const User = require("./User");
 * @prop {String?} messageReference.messageID The id of the original message this message was crossposted from
 * @prop {String} messageReference.channelID The id of the channel this message was crossposted from
 * @prop {String?} messageReference.guildID The id of the guild this message was crossposted from
+* @prop {Object?} interaction An object containing the interaction for the original message if it has been created by an interaction
+* @prop {String} interaction.id The id of the interaction the message was created from
+* @prop {String} interaction.type The type of interaction the message was created from
```
Comment (bsian03) [inline lib/structures/Message.js]:
```suggestion
* @prop {Number} interaction.type The type of interaction the message was created from
```

File: index.d.ts [hunk 3]
```diff
@@ -797,6 +798,13 @@ declare namespace Eris {
     preview_asset?: string;
     format_type: Constants["StickerFormats"][keyof Constants["StickerFormats"]];
   }
+  interface MessageInteraction {
+    id: string;
+    type: InteractionType;
+    name: string;
+    user: User;
+    member?: Member;
```
Comment (bsian03) [inline index.d.ts]:
```suggestion
    member: Member | null;
```

File: index.d.ts [hunk 4]
```diff
@@ -2124,6 +2132,7 @@ declare namespace Eris {
     referencedMessage?: Message | null;
     roleMentions: string[];
     stickers?: Sticker[];
+    interaction?: MessageInteraction;
```
Comment (bsian03) [inline index.d.ts]:
```suggestion
    interaction: MessageInteraction | null;
```

**Review by bsian03 — CHANGES_REQUESTED** [review]

**Review by abalabahaha — DISMISSED** [review]

File: lib/structures/Message.js [hunk 5]
```diff
@@ -31,6 +31,11 @@ const User = require("./User");
 * @prop {String?} messageReference.messageID The id of the original message this message was crossposted from
 * @prop {String} messageReference.channelID The id of the channel this message was crossposted from
 * @prop {String?} messageReference.guildID The id of the guild this message was crossposted from
+* @prop {Object?} interaction An object containing the interaction for the original message if it has been created by an interaction
+* @prop {String} interaction.id The id of the interaction the message was created from
+* @prop {Number} interaction.type The type of interaction the message was created from
+* @prop {User} interaction.user The user who invoked the interaction the message was created from
+* @prop {Member?} interaction.member The member who invoked the interaction the message was created from
```
Comment (abalabahaha) [inline lib/structures/Message.js]:
```suggestion
* @prop {Object?} interaction An object containing info about the interaction the message is responding to, if applicable
* @prop {String} interaction.id The id of the interaction
* @prop {Number} interaction.type The type of interaction
* @prop {User} interaction.user The user who invoked the interaction
* @prop {Member?} interaction.member The member who invoked the interaction
```

**Review by abalabahaha — APPROVED** [review]

**Pull request merged**